src/rkopenmdao/checkpoint_interface/all_checkpointer.py

```diff
@@ -27,16 +27,20 @@ def iterate_forward(self, initial_state: np.ndarray):
         old_delta_t = None
         old_norm = None
         while self.integration_control.termination_condition_status():
-            self._storage.append((self._state.copy(), self.integration_control.delta_t, old_delta_t, old_norm))
-            self._state, old_delta_t, old_norm = self.run_step_func(self._state.copy())
-            self.integration_control.increment_step()
+            self._storage.append([self._state.copy(),0.0 ,0.0])
```

Comment:
Please add a small comment explaining why this structure for the checkpoint storage is necessary. While it is clear right to us right now, it might look strange for others reading the code (or possibly also for us a few months down the line).